### ISSUE TITLE
how to adjust the position of quickhelp

### ISSUE THREAD

**Issue by windrg** [expez/company-quickhelp#52]
Hi,

This package seems to locate its quickhelp in the right middle of its completion list like this.

![screenshot from 2016-10-24 22-04-22](https://cloud.githubusercontent.com/assets/506484/19646940/ccc48972-9a36-11e6-8252-02862ebe4159.png)

I'd like to adjust it up to the top right of its completion list like as eldoc in AC package.
Could you please let me know how to do it?

**Comment by expez** [comment]
What you're seeing is a bug, it's not supposed to work like that.  Are you on OSX? If so, you too might be suffering from [this bug](https://github.com/pitkali/pos-tip/issues/11).

If you want to mess around with this yourself, [these](https://github.com/expez/company-quickhelp/blob/master/company-quickhelp.el#L144-L147) are the variables to twiddle.

**Comment by quicknir** [comment]
@expez I'm not quite seeing that bug, but ultimately I came here to request the same thing: some kind of setting to control where the tool tip appears. In my case, the problem is that I'm working with emacs on the left of my two monitors. When working with a pane in the right of emacs, the tooltip decides that since there's more space available, it's ok to span the tooltip onto my second monitor. This makes it hard to read the tooltip in practice. When emacs is on my right monitor, the tooltip has nowhere to shift so far right so it does something more reasonable.

This could be solved either by a setting that allows treating the monitor as a hard boundary (which may be difficult), or by having some kind of variable that controls the relative tooltip position. For instance, this may sound strange to some but I'd be happy to put the tooltip slightly above the text I'm completing.

**Comment by expez** [comment]
@quicknir Yeah, that would be cool.  

That feature belongs in `pos-tip`, though, which is reponsible for displaying the popup.  All I do is go "Display this text in a popup near this position.".

I've delegated this stuff to `pos-tip`, because I'm no glutton for punishment.  It's really hard work to maintain solid cross-platform behavior with regard to popups.

**Comment by quasilyte** [comment]
Same issue on open suse with emacs 25.1.1.

**Comment by nenadalm** [comment]
Hi. I just installed the plugin and my problem is that company-quickhelp puts the documentation in place of autocomplete (so I don't see list of available functions) in case [neotree][0] is opened. If [neotree][0] is closed - it seems to work well.

[0]: https://www.emacswiki.org/emacs/NeoTree

**Comment by azzamsa** [comment]
![img](https://i.redd.it/6qq0zxds06a11.png)

company-box locate the 'quickhelp' nicely. Can we have the same thing ?

1) there is little space between company tooltip and 'quickhelp'
2) the quickhelp doesn't have line border

Thank you

**Comment by expez** [comment]
Yes, that might be possible as an alternative to `pos-tip`.  That would mean dropping support for all emacs < 26 AFAICT.  Not sure that would be a good idea just yet.

Thanks for bringing that library to my attention!

**Comment by dgutov** [comment]
@azzamsa Any particular reason you didn't just want to use `company-box`?

If it's about its lack of some "standard" features, you can try https://github.com/tumashu/company-posframe/ now. That one includes quickhelp functionality as well.

**Comment by dgutov** [comment]
> there is little space between company tooltip and 'quickhelp'

This one seems easy enough to add. What distance would people like?

**Comment by azzamsa** [comment]
@dgutov missing support for number. But there is a patch for that now. I will give it second try.
